src/core/mac/mac.cpp

```diff
@@ -578,6 +582,10 @@ void Mac::NextOperation(void)
 
 void Mac::ScheduleNextTransmission(void)
 {
+    VerifyOrExit(!mWaitingForData);
```

Comment:
We should set `mWaitngForData` to `true` from somewhere (right now it will always remain `false`). We should do this after sending a data poll and getting an ack with frame pending bit set (parent telling us it has something for us). I think we can add `mWaitingForData = true;` right after `mReceiveTimer.Start(kDataPollTimeout)` in either one of the `Mac::TransmitDoneTask()`s (we have two to handle the legacy ack model).